File: Tercero/Desarrollo Software-DS/Practica 1/Sesion3/Obligatoria/S3/ManagerFilter.cpp
#include "ManagerFilter.h"

ManagerFilter::ManagerFilter(Tecnico * t)
{
    this->tec=t;
    cf=new CadenaFiltro();
    
    crearCadena();
    
}

void ManagerFilter::crearCadena(){
    
    cf->generar();
    
}    

void ManagerFilter::operacionFiltrar(){
    cf->filtrar(tec->getTrabajo());
    
}

File: Desarrollo Software-DS/Practica 1/Sesion3/Opcional/dart/lib/managerFilter.dart
import 'cadenaFiltro.dart';
import 'tecnico.dart';

class ManagerFilter{
  Tecnico  tec= new Tecnico("", 0);
  CadenaFiltro cf= new CadenaFiltro();

  ManagerFilter(Tecnico t){
    this.tec=t;
    this.cf=new CadenaFiltro();
    
    crearCadena();
  }

  void crearCadena(){
      cf.generar();
  }

  void operacionFiltrar(){
      cf.filtrar(tec.getTrabajo());
  }

}